engines/order_management/app/services/order_management/stock/package.rb

```diff
@@ -27,13 +92,40 @@ def shipping_categories
       #
       # @return [Array<Spree::ShippingMethod>]
       def shipping_methods
-        available_shipping_methods = super.to_a
+        available_shipping_methods = shipping_categories.map(&:shipping_methods).flatten.uniq.to_a
```

Comment:
we can replace `#map` + `#flatten` with [#flat_map](https://apidock.com/ruby/v2_2_9/Enumerable/flat_map). have you checked whether this introduces an N+1 though? I have no idea if at this point that shipping methods are already in memory.

Reply:
I was trying to avoid changes that are not straight forward to avoid introducing risk. Anyway, I did this one now :+1: